apiServices/fetchDiet/index.js

```diff
@@ -1,23 +1,21 @@
 const url = "/api/ninjas/";
 
-export default async function fetchDiet(
-  input,
-  setFetchingStatus,
-  setSearchResults
-) {
+export default async function fetchDiet(input, setFetchingStatus) {
   setFetchingStatus("Fetching Data");
   try {
     const response = await fetch(url + input);
     if (response.ok) {
       const dataFetch = await response.json();
-      setSearchResults(dataFetch);
       setFetchingStatus("Fetching Done");
+      return dataFetch;
     } else {
       setFetchingStatus("Error");
       console.error("Response not OK.");
+      return "Error";
```

Comment:
You should remove the `setFetchingStatus` callback as well and instead throw an error in case something goes wrong. You should then catch the error on the calling side and call `setFetchingStatus` there.